clicommand/agent_start.go

```diff
@@ -1257,6 +1270,28 @@ var AgentStartCommand = cli.Command{
 	},
 }
 
+// checkBinaryPaths looks up both the bootstrap and host buildkite-agent paths,
+// and returns an error if they do not match or if either cannot be determined.
+func checkBinaryPaths() error {
+	bootstrapPath, err := exec.LookPath("buildkite-agent")
+	if err != nil {
+		return fmt.Errorf("failed to locate bootstrap buildkite-agent: %w", err)
+	}
+
+	hostPath, err := os.Executable()
```

Comment:
The documentation points out this may return a symlink, which could cause inconsistencies. It recommends evaluating the symlink:

> If a symlink was used to start the process, depending on the operating system, the result might be the symlink or the path it pointed to. If a stable result is needed, [path/filepath.EvalSymlinks](https://pkg.go.dev/path/filepath#EvalSymlinks) might help.

Reply:
Thanks for looking that up, good catch! I've updated it to handle symlinks.